File: docs/examples/py/src/data_management_delete_query.py
from time import time

from reduct import Client, Bucket, Batch


async def main():
    # Create a client with the base URL and API token
    async with Client("http://localhost:8383", api_token="my-token") as client:
        bucket: Bucket = await client.create_bucket("my-bucket", exist_ok=True)

        # Send some records to the "py-example"
        ts = time()
        await bucket.write(
            "py-example",
            b"Some binary data",
            ts
        )
        await bucket.write(
            "py-example",
            b"Some binary data",
            ts + 1
        )

        # Delete all records within a time range
        removed_records = await bucket.remove_query("py-example", ts, ts + 2)
        assert removed_records == 2

        # You can also delete all records with a specific label
        await bucket.remove_query("py-example", ts, ts + 2, when={"&key1": {"$eq": "value1"}})

        # Or each N-th record
        await bucket.remove_query("py-example", ts, ts + 2, each_n=2)

if __name__ == "__main__":
    import asyncio

    asyncio.run(main())

File: versioned_docs/version-1.21.x/examples/go/src/data_management_delete_query.go
package main

import (
	"context"
	reduct "github.com/reductstore/reduct-go"
	"time"
)

func main() {
	// Create a client and use the base URL and API token
	client := reduct.NewClient("http://localhost:8383", reduct.ClientOptions{
		APIToken: "my-token",
	})

	// Get or create a bucket with the name "my-bucket"
	bucket, err := client.CreateOrGetBucket(context.Background(), "my-bucket", nil)
	if err != nil {
		panic(err)
	}

	// Send some records to the "go-example"
	ts := time.Now().UnixMicro()
	err = bucket.BeginWrite(context.Background(), "go-example", &reduct.WriteOptions{
		Timestamp: ts,
	}).Write([]byte("Some binary data"))
	if err != nil {
		panic(err)
	}

	err = bucket.BeginWrite(context.Background(), "go-example", &reduct.WriteOptions{
		Timestamp: ts + 1,
	}).Write([]byte("Some binary data"))
	if err != nil {
		panic(err)
	}

	// Delete all records within a time range
	queryRequest := &reduct.QueryOptions{
		Start: ts,
		Stop:  ts + 2,
	}
	removedRecords, err := bucket.RemoveQuery(context.Background(), "go-example", queryRequest)
	if err != nil {
		panic(err)
	}
	if removedRecords != 2 {
		panic("Expected to remove 2 records, but got " + string(removedRecords))
	}

	// You can also delete all records with a specific label
	_, err = bucket.RemoveQuery(context.Background(), "go-example", &reduct.QueryOptions{
		Start: ts,
		Stop:  ts + 2,
		When:  map[string]any{"&key1": map[string]any{"$eq": "value1"}},
	})

	if err != nil {
		panic(err)
	}

	// Or each second record in a time range
	_, err = bucket.RemoveQuery(context.Background(), "go-example", &reduct.QueryOptions{
		Start: ts,
		Stop:  ts + 2,
		When:  map[string]any{"$each_n": 2},
	})

	if err != nil {
		panic(err)
	}
}
